javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -36,11 +60,10 @@
     }
 
     static Object[] copy(Object arrayObject, int minLength) {
-        final Object[] array = (Object[]) arrayObject;
-        final int arrayLength = (array == null) ? 0 : array.length;
-        final int length = Math.max(arrayLength, minLength);
-        return (length == arrayLength) ? array.clone()
-                                       : arrayCopy(length, array, 0, 0, arrayLength);
+        final Object[] array = (arrayObject == null) ? EMPTY : (Object[]) arrayObject;
+        return (array.length >= minLength)
```

Comment:
What if array == EMPTY? Would it make sense to just return EMPTY then instead of calling `close()` (can minLength == 0?) or `arrayCopy()`?